src/plays/pokemon-stats/components/Search.tsx

```diff
@@ -1,6 +1,22 @@
 import React from "react";
 import "./search.css";
-const search = (props) => {
+type Propsi = {
+  setname: React.Dispatch<React.SetStateAction<string>>;
+  name: string;
+  fetchPokemonDetails: (e: React.FormEvent<HTMLFormElement>) => void;
+};
+
+type Props = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  type?: "submit";
+};
```

Comment:
@yung-coder , We can use the interface here and extend the `ButtonHTMLAttributes`.
Also, we don't need to define the `type` explicitly, `ButtonHTMLAttribute` already provide the type like this.
```
type?: 'submit' | 'reset' | 'button' | undefined;
```
You can create your prop like this.
```ts
interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {}
```